### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by vitormattos**
Hi @eneiluj ,
I'm reviewing your app's code. I want to improve performance and parallelize the download of Dropbox files.

Is it okay for you to send separate pull requests to make it easier to review the code?

In this pull request I removed an unused property and added the service property to improve coding in an IDE.

### CODE REVIEW

**Comment by eneiluj** [comment]
Hi @vitormattos,

As long as the changes don't mix multiple different things/topics, you can make big PRs, i don't mind.
Do you want to add stuff to this PR or can I merge it right now?

**Comment by eneiluj** [comment]
And by the way, as Dropbox most likely sends faster than everybody can receive, I don't feel it's necessary to parallelize file download as you're still going to get stuck at your connection max speed.

**Comment by vitormattos** [comment]
For now nothing to add to it.

Because the app is without unit tests, I'm doing manual tests. I think it's safer to go for smaller parts so as not to get too complex when doing the review.

**Comment by vitormattos** [comment]
I need to transfer more than 600 GB from Dropbox to the Nextcloud and for some reason the job is dying without logs after about 60 GB transferred.

Importing files has other processes besides downloading, for example IO. When the import is done in a queue and for many files, the time increases.

The goal is to parallel a poll of up to 5 requests so that, while a download is being made, a response promise from another download is being processed.

**Comment by vitormattos** [comment]
Just in time, I can put this pull request as a draft and add everything to it. Finally, you do one last review.
Do you prefer it this way?

Now that I recognize you, hi @eneiluj !!!

**Comment by vitormattos** [comment]
Hi @eneiluj ,
I made some refactors and I'm testing with the import of 600Gb, it is without problems.
I suggest you do the code review by the sequence of commits.

File: lib/Service/DropboxStorageAPIService.php [hunk 1]
```diff
@@ -247,48 +238,49 @@ private function getFile(string $accessToken, string $refreshToken, string $clie
...
 					['app' => $this->appName]
 				);
 				return null;
 			}
-			$res = $this->dropboxApiService->downloadFile(
-				$accessToken, $refreshToken, $clientID, $clientSecret, $userId, $resource, $fileItem['id']
+		}
+		if ($saveFolder->nodeExists($fileName)) {
+			return null;
+		}
```
Comment (eneiluj) [inline lib/Service/DropboxStorageAPIService.php:251]:
```suggestion
		try {
			if ($saveFolder->nodeExists($fileName)) {
				return null;
			}
		} catch (ForbiddenException $e) {
			return null;
		}
```
In some cases, this is needed. For example when importing files named `.htaccess`.

**Review by eneiluj — CHANGES_REQUESTED** [review]
Thanks for this PR. A few things to adjust and let's merge it.

File: lib/Service/DropboxStorageAPIService.php [hunk 2]
```diff
@@ -238,7 +229,7 @@ public function importFiles(string $accessToken, string $refreshToken, string $c
 	 * @return ?int downloaded size, null if already existing or network error
 	 */
 	private function getFile(string $accessToken, string $refreshToken, string $clientID, string $clientSecret,
-							string $userId, array $fileItem, Node $topFolder): ?float {
+							string $userId, array $fileItem, Node $topFolder): ?int {
```
[the remark this answers is not in the document]
Comment (vitormattos) [reply lib/Service/DropboxStorageAPIService.php]: It’s good to have unit tests to document these exception scenarios.

I did tests with files over 1024Mb. An error occurs that I have not been able to resolve. The error occurs with the code before and after this my PR. If you want I can register an issue with the log.

[the remark this answers is not in the document]
Comment (eneiluj) [reply lib/Service/DropboxStorageAPIService.php]: > If you want I can register an issue with the log.

Yes sure :grin:.

**Comment by eneiluj** [comment]
You forgot to import `OCP\Files\ForbiddenException` :wink:.

**Comment by vitormattos** [comment]
@eneiluj done.

**Review by eneiluj — APPROVED** [review]
Thanks!

**Pull request merged**